### PR TITLE
DAOS-4365 profile: Add aggregate option to profile

### PR DESCRIPTION
**Pull request by wangdi1**
Add aggregate/average option to profile.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

File: src/include/daos_srv/daos_server.h [hunk 1]
```diff
@@ -123,9 +123,11 @@ do {						\
 
 #define D_TIME_END(sp, start, op)		\
 do {						\
+	int time_msec;				\
 	if ((sp) == NULL || start == 0)		\
 		break;				\
-	srv_profile_count(sp, op, (int)(daos_get_ntime() - start)); \
+	time_msec = (daos_get_ntime() - start)/1000; \
```
Comment (NiuYawei) [inline src/include/daos_srv/daos_server.h]: time_usec?

**Comment by NiuYawei** [comment]
BTW, it'll be great to compute the std deviation at the same time.

File: src/iosrv/profile.c [hunk 2]
```diff
@@ -58,37 +67,31 @@ srv_profile_chunk_alloc(int chunk_size)
 		return NULL;
 
 	D_INIT_LIST_HEAD(&chunk->spc_chunk_list);
-
-	D_ALLOC(chunk->spc_profiles,
-		chunk_size * sizeof(*chunk->spc_profiles));
-	if (chunk->spc_profiles == NULL) {
-		D_FREE_PTR(chunk);
+	D_ALLOC(chunk->spc_chunks, chunk_size);
```
Comment (liuxuezhao) [inline src/iosrv/profile.c]: shouldn't it be D_ALLOC(chunk->spc_chunks, chunk_size * sizeof(*chunk->spc_chunks));?

Comment (liuxuezhao) [reply src/iosrv/profile.c]: > shouldn't it be D_ALLOC(chunk->spc_chunks, chunk_size * sizeof(*chunk->spc_chunks));?

@wangdi1 is it a bug actually? or may I missunderstood it?

Comment (wangdi1) [reply src/iosrv/profile.c]: Ah, yes, you are right. I will fix it in the following patch.

**Pull request merged**